core/src/main/java/nl/nn/adapterframework/pipes/FixedResultPipe.java

```diff
@@ -131,57 +129,55 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 				PipeForward fileNotFoundForward = findForward(FILE_NOT_FOUND_FORWARD);
 				if (fileNotFoundForward != null) {
 					return new PipeRunResult(fileNotFoundForward, message);
-				} else {
-					throw new PipeRunException(this,getLogPrefix(session)+"cannot find resource ["+filename+"]");
 				}
+				throw new PipeRunException(this,getLogPrefix(session)+"cannot find resource ["+filename+"]");
 			}
 			try {
 				result = Misc.resourceToString(resource, Misc.LINE_SEPARATOR);
 			} catch (Throwable e) {
 				throw new PipeRunException(this,getLogPrefix(session)+"got exception loading ["+filename+"]", e);
 			}
 		}
-		if (getParameterList()!=null) {
-			ParameterValueList pvl;
+		if (StringUtils.isNotEmpty(getReplaceFrom()) && result != null) {
+			result = replace(result, getReplaceFrom(), getReplaceTo());
+		}
+		if (!getParameterList().isEmpty()) {
 			try {
-				pvl = getParameterList().getValues(message, session);
+				ParameterValueList pvl = getParameterList().getValues(message, session);
+				for(ParameterValue pv : pvl) {
+					String replaceFrom;
+					if (isReplaceFixedParams()) {
+						replaceFrom=pv.getName();
+					} else {
+						replaceFrom="${"+pv.getName()+"}";
+					}
+					result=replace(result, replaceFrom, pv.asStringValue(""));
+				}
 			} catch (ParameterException e) {
 				throw new PipeRunException(this,getLogPrefix(session)+"exception extracting parameters",e);
 			}
-			for(ParameterValue pv : pvl) {
-				String replaceFrom;
-				if (isReplaceFixedParams()) {
-					replaceFrom=pv.getDefinition().getName();
-				} else {
-					replaceFrom="${"+pv.getDefinition().getName()+"}";
-				}
-				result=replace(result,replaceFrom,pv.asStringValue(""));
-			}
 		}
 
 		message.closeOnCloseOf(session, this); // avoid connection leaking when the message itself is not consumed.
-		if (getSubstituteVars()){
-			result=StringResolver.substVars(returnString, session, appConstants);
+		if (isSubstituteVars()) {
+			result=StringResolver.substVars(result, session, appConstants);
 		}
 
-		if (StringUtils.isNotEmpty(styleSheetName)) {
-			URL xsltSource = ClassUtils.getResourceURL(this, styleSheetName);
+		if (StringUtils.isNotEmpty(getStyleSheetName())) {
+			URL xsltSource = ClassUtils.getResourceURL(this, getStyleSheetName());
 			if (xsltSource!=null) {
 				try{
-					String xsltResult = null;
 					Transformer transformer = XmlUtils.createTransformer(xsltSource);
-					xsltResult = XmlUtils.transformXml(transformer, result);
-					result = xsltResult;
+					result = XmlUtils.transformXml(transformer, result);
```

Comment:
It would perform a whole lot better if you were to use a transformerpool  -hint hint- 😄